ui/src/ui-utils.ts

```diff
@@ -64,3 +64,62 @@ export function truncateText(text: string, limit: number, withEllipsis: boolean
     return text;
   }
 }
+
+/**
+ * Get the correct display name for certain publishers.
+ * 
+ * @param {string} the type of primaryPaperLInk
+ */
+export function getLinkText(input: string): string {
+
+  var linkText: string = "";
+
+  if (input.length > 17) {
```

Comment:
can you give 17 a descriptive constant name, describing what the value means?

```suggestion

  const <DESCRIPTIVE_CONSTANT_NAME> = 17;
  if (input.length > <DESCRIPTIVE_CONSTANT_NAME>) {
```